### PR TITLE
feat(jcasc): add additionalExistingSecrets capability

### LINKED ISSUE #334: Use existing secrets
**Is your feature request related to a problem? Please describe.**
The manual for using a secret in JCasC is there: https://artifacthub.io/packages/helm/jenkinsci/jenkins/3.3.6#additional-secrets. But not that perfect in my opinion (just to think about) ;)

**Describe the solution you'd like**
On the one hand there should be mentioned that the way to `secrets can be mounted into the Jenkins controller through the chart` is the `persistence.mount` or what it is called.. But the better way would be to do the same as for `additionalSecrets`  but for `existingSecrets` where I just have to put the secret name which already exists. Easier to find and I am not sure if `persistence.enabled` must be true to use `persistence.mounts` .. 
(How the JCasC knows those keys as they are just mounted anywhere I don't know yet)

**Describe alternatives you've considered**
Just put the note about `persistence.mount` in the Chart Readme (but I would prefer the `existingSecret` way :) )

**Additional context**

### PR DESCRIPTION
**Pull request by ArieLevs**
@maorfr @torstenwalter @mogaal 

# What this PR does / why we need it
adds the ability to mount existing secrets to use with [kubernetes secrets jcasc](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/features/secrets.adoc#kubernetes-secrets)

# Which issue this PR fixes

- fixes #334(, fixes #334)
# Special notes for your reviewer
the [Additional Secrets](https://github.com/jenkinsci/helm-charts/tree/main/charts/jenkins#additional-secrets) feature by itself does not resolve the issue of providing sensitive info to the chart.  
since the `value` key expects a sensitive secret, it cannot commit to git, thus the problem moved from `JCasC` property to `additionalSecrets`.
this feature aims to resolve the issue above.

# Checklist

- [x] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] CHANGELOG.md was updated

### CODE REVIEW

File: charts/jenkins/README.md [hunk 1]
```diff
@@ -341,12 +341,24 @@ If the storage class is set to null or left undefined (`""`), the default provis
...
 controller:
+  # 'name' and 'keyName' must be lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-',
+  # and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc')
+  additionalExistingSecrets:
+    - name: secret-credentials
+      keyName: github-username
+    - name: secret-credentials
+      keyName: github-password
+   - name: secret-credentials
+      keyName: token
```
Comment (torstenwalter) [inline charts/jenkins/README.md:360]:
```suggestion
   - name: secret-credentials
     keyName: token
```

Comment (torstenwalter) [reply charts/jenkins/README.md:360]: Without looking at the configScripts example I would not know that `name` and `keyName` are concatenated                   `secret: ${secret-credentials-token}`.
I thinks that's worth documenting here.

File: charts/jenkins/VALUES_SUMMARY.md [hunk 2]
```diff
@@ -226,6 +226,7 @@ The following tables list the configurable parameters of the Jenkins chart and t
 | `controller.adminUser`                | Admin username (and password) created as a secret if adminSecret is true | `admin` |
 | `controller.adminPassword`            | Admin password (and user) created as a secret if adminSecret is true | Random value |
 | `controller.additionalSecrets`        | List of additional secrets to create and mount according to [JCasC docs](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/features/secrets.adoc#kubernetes-secrets) | `[]` |
+| `controller.additionalExistingSecrets`| List of additional existing secrets to create and mount according to [JCasC docs](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/features/secrets.adoc#kubernetes-secrets) | `[]` |
```
Comment (torstenwalter) [inline charts/jenkins/VALUES_SUMMARY.md:229]:
```suggestion
| `controller.additionalExistingSecrets`| List of additional existing secrets to mount according to [JCasC docs](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/docs/features/secrets.adoc#kubernetes-secrets) | `[]` |
```
Those secrets need to already exist. That's also what `controller.additionalExistingSecrets` indicates.

**Review by torstenwalter — CHANGES_REQUESTED** [review]

File: charts/jenkins/README.md (hunk 1 above)
Comment (ArieLevs) [reply charts/jenkins/README.md]: readme and comment inside values file updated

File: charts/jenkins/VALUES_SUMMARY.md (hunk 2 above)
Comment (ArieLevs) [reply charts/jenkins/VALUES_SUMMARY.md]: thanks, I've copied the row from `additionalSecrets` and missed the create part, updated

**Comment by ArieLevs** [comment]
@torstenwalter Hi can you please re-review

**Comment by torstenwalter** [comment]
looks good to me. Can you resolve the conflicts one more time?
Bonus points if you add a helm unit test to check if everything is rendered as you expect it so that we can't break it in the future.

**Comment by ArieLevs** [comment]
@torstenwalter conflicts resolved,
regarding unit tests, I wasn't not able to test them locally.
only `ct lint` was ok, by executing, for example:
```
docker run -it --rm -v "${PWD}:/charts" --workdir /charts quay.io/helmpack/chart-testing:v3.1.1 ct lint --config ct.yaml
```

but having errors if running `ct install` command, is there a contribution guide on how to run them?

**Comment by torstenwalter** [comment]
First you need to install the helm unit test plugin:
https://github.com/jenkinsci/helm-charts/blob/main/.github/workflows/lint-test.yaml#L53

The you can run it:
https://github.com/jenkinsci/helm-charts/blob/main/ct.yaml#L8

**Comment by ArieLevs** [comment]
I've tried running the exact command couple days ago, getting errors like
```shell
### Error:  file 'tests/*.yaml' seems to be a YAML file, but expected a gzipped archive
```
after exploring this tool a bit, I managed to execute the tests using the [documentation](https://github.com/quintush/helm-unittest#test-suite-file), but it actually only ran a single test, [this one](https://github.com/jenkinsci/helm-charts/blob/main/charts/jenkins/tests/jenkins-controller-servicemonitor_test.yaml) since its the only one contained an underscore in its name (`*_test.yaml`), not sure if its an issue, but anyway, I've wrote the test but pushed it with a name similar to most other tests (`*-test.yaml`).

test passing locally, please let me know if its covering enough.
@tobilarscheid

**Review by torstenwalter — APPROVED** [review]

**Comment by torstenwalter** [comment]
@ArieLevs Thank you for adding this and including tests!

**Pull request merged**